code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -950,22 +950,38 @@
 	icon = 'icons/obj/medical/chemical.dmi'
 	icon_state = "potlightpink"
 
-/obj/item/slimepotion/genderchange/attack(mob/living/L, mob/user)
-	if(!istype(L) || L.stat == DEAD)
+/obj/item/slimepotion/genderchange/attack(mob/living/gendered_thing, mob/user)
+	if(!istype(gendered_thing) || gendered_thing.stat == DEAD)
 		to_chat(user, span_warning("The potion can only be used on living things!"))
 		return
 
-	if(L.gender != MALE && L.gender != FEMALE)
+	if(gendered_thing.gender != MALE && gendered_thing.gender != FEMALE)
 		to_chat(user, span_warning("The potion can only be used on gendered things!"))
 		return
+	if(gendered_thing.mind && gendered_thing != user)
+		to_chat(user, span_notice("You offer the [name] to [gendered_thing]."))
+		if(tgui_alert(gendered_thing, "Do you accept the [name], knowing that it will permanently change your gender?", "Gender Changer", list("Yes", "No")) == "Yes")
```

Comment:
```suggestion
		if(tgui_alert(gendered_thing, "Do you accept [src], knowing that it will permanently change your gender?", "Gender Changer", list("Yes", "No")) == "Yes")
```